test/presubmit-tests.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs the presubmit tests; it is started by prow for each PR.
+
+function build_tests() {
+  echo "Running build tests"
+  make generate
+}
+
+function unit_tests() {
+  echo "Running unit tests"
+  make test
+}
+
+function integration_tests() {
+  header "Running integration tests"
+  echo "TODO(aaron-prindle): add integration tests"
+}
+
+# Process flags and run tests accordingly.
+function main() {
```

Comment:
You're unnecessarily duplicating code that is now intended to be vendored. I strongly recommend that you follow this pattern instead:

https://github.com/knative/docs/blob/master/test/presubmit-tests.sh